website/source/docs/guides/connect-gateways.md

```diff
@@ -0,0 +1,393 @@
+`token` options. This is what you would add to an existing service registration
+if you are not using socat as an example.
+
+Reload the client with the new or modified registration.
+
+```text
+$ consul reload
+```
+
+Then start Envoy specifying which service it will proxy.
+
+```text
+$ consul connect envoy -sidecar-for socat
+```
+
+If you are using socat as your example, start it now by running the following
+command.
+
+```text
+$ socat -v tcp-l:8181,fork exec:"/bin/cat"
+```
+
+Check that the socat service is running by accessing it using netcat on the same
+node. It will echo back anything you type.
+
+```text
+$ nc 127.0.0.1 8181
+hello
+hello
+echo
+echo
+```
+
+Stop the running netcat service by typing `ctrl + c`.
+
+### Register a front end service in the other datacenter
+
+Now in your other datacenter, you will register a service (with a sidecar proxy)
+that calls your backend service. Your registration will need to list the backend
+service as your upstream. Like the backend service, you can use an example
+service, which will be called web, or append the connect stanza to an existing
+registration with some customization.
+
+To use web as your front end service, create a registration file called
+`web.json` that contains the following snippet.
+
+```json
+{
+  "service": {
+    "name": "web",
+    "port": 8080,
+    "token": "<token here>",
+    "connect": {
+      "sidecar_service": {
+        "proxy": {
+          "upstreams": [{
+            "destination_name": "socat",
+            "datacenter": "primary",
+            "local_bind_port": 8181
+          }]
+        }
+      }
+    }  
+  }
+}
+```
+
+Note the Connect part of the registration, which specifies socat as an
+upstream. If you are using another service as a back end, replace `socat` with
+its name and the `8181` with its port.
+
+Reload the client with the new or modified registration.
+
+```text
+$ consul reload
+```
+
+Then start Envoy and specify which service it will proxy.
+
+```text
+$ consul connect envoy -sidecar-for web
+```
+
+## Configure Intentions to Allow Communication Between Services
+
+Now that your services both use Connect, you will need to configure intentions
+in order for them to communicate with each other. Add an intention to allow the
+front end service to access the back end service. For web and socat the command
+would look like this.
+
+```text
+$ consul intention create web socat
+```
+
+Consul will automatically forward intentions initiated in the in the secondary
+datacenter to the primary datacenter, where the servers will write them. The
+servers in the primary datacenter will then automatically replicate the written
+intentions back to the secondary datacenter.
+
+## Test the connection
+
+Now that you have services using Connect, verify that they can contact each
+other. If you have been using the example web and socat services, from the node
+and datacenter where you registered the web service, start the socat service
```

Comment:
This is almost there, needs one more adjustment. The socat service is already running upstream, so to test web's connection to socat we use netcat:
>If you have been using the example web and socat services, from the node and datacenter where you registered the web service, use netcat to test the connection to the socat echo server.

Reply:
Good catch, thank you!